### PR TITLE
Fix storybook and chromatic errors

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr updates some components and stories to resolve storybook / chromatic issues. Therefore the Deck and CardList component are wrapped into new components that handle fetching the data.

### CODE REVIEW

File: frontend/components/deck/Deck.tsx [hunk 1]
```diff
@@ -82,10 +87,10 @@ export default function Deck({ group, deck, className = "" }: DeckProps) {
 							<AlertTriangle size={50}></AlertTriangle>
 						)}
 				</div>
-				{dueCards > 0 && (
+				{!!deck?.dueCards && deck.dueCards > 0 && (
```
Comment (darmiel) [inline frontend/components/deck/Deck.tsx:90]:
```suggestion
				{deck?.dueCards > 0 && (
```

File: frontend/components/flashcard/CardList.tsx [hunk 2]
```diff
@@ -22,19 +26,28 @@ interface CardListProps {
...
- * UI component for displaying a list of cards
- */
-export const CardList = ({ deck, setCard, className = "" }: CardListProps) => {
+export const FetchCardList = ({ deck, ...props }: CardListProps) => {
 	const fetcher = (url: RequestInfo | URL) =>
 		authedFetch(url, {
 			method: "GET",
 		}).then((res) => res?.json());
 	const { data: cards } = useSWR(`/api/decks/${deck.deckID}/cards`, fetcher);
+	return <CardList deck={deck} cards={cards?.cards} {...props}></CardList>;
```
Comment (darmiel) [inline frontend/components/flashcard/CardList.tsx:35]:
```suggestion
	return <CardList deck={deck} cards={cards?.cards} {...props} />;
```

File: frontend/components/navigation/Tabs/CardsTab.tsx [hunk 3]
```diff
@@ -29,11 +29,11 @@ export const CardsTab = ({ deck, className = "" }: CardsTabProps) => {
 		<div
 			className={`flex h-full max-h-full flex-col md:flex-row ${className}`}
 		>
-			<CardList
+			<FetchCardList
 				deck={deck}
 				setCard={setNewCard}
 				className={`${card ? "md:w-1/2" : "w-full"}`}
-			></CardList>
+			></FetchCardList>
```
Comment (darmiel) [inline frontend/components/navigation/Tabs/CardsTab.tsx:36]:
```suggestion
			/>
```

**Review by darmiel — CHANGES_REQUESTED** [review]

**Review by TomRomeo — CHANGES_REQUESTED** [review]
I am not really happy with the naming. I would rather have the component with the logic have the simple name (eg. `Deck`) and the purely visual component to have another name (eg. `DeckVisual`) since developers will mostly use the components with logic in other parts of the program.

**Review by darmiel — APPROVED** [review]

**Review by memeToasty — APPROVED** [review]

**Review by 4KevR — APPROVED** [review]

**Review by TomRomeo — APPROVED** [review]
Really unhappy with the naming but since the majority seems to like it... 😭

**Pull request merged**